src/mono/mono/metadata/bundled-resources.c

```diff
@@ -0,0 +1,140 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+//
+
+#include <glib.h>
+#include <stdbool.h>
+
+#include "bundled-resources-internals.h"
+#include "assembly-internals.h"
+
+#include <mono/metadata/assembly.h>
+#include <mono/metadata/mono-private-unstable.h>
+
+static GHashTable *bundled_resources = NULL;
+static gboolean bundle_contains_assemblies, bundle_contains_satellite_assemblies = false;
```

Comment:
If its gboolean, use TRUE/FALSE, if you use C99 bool, then use true/false.

Reply:
In runtime code, we historically use gboolean in Mono, since bool is C99 and before we adopted that, there was not bool type as part of the C standard. gboolean is typed to a 4 byte integer, so if you need a known size of the boolean, then gboolean is explicit about that. Majority of runtime code uses gboolean, so in order to be inline with existing runtime code, ou can use gboolean. Code shared between runtimes (like code under src/native) adapts C99 data types and use bool. Keep in mind that conversion between gboolean and bool is not automatically portable, that's why you see things like !! when converting from gboolean to bool. In interop code we also have MonoBoolean that is always typed to a 1 byte data type, so its 1 byte in size, that is needed to be in sync with C# bool.

Reply:
The `EmitBundle` bundling task emits a header file (`mono-bundled-source.h`) that is essentially the same as `bundled-resources-internals.h`. This header file (`mono-bundled-source.h`) is what would eventually be compiled into the source/object files that end up bundled into the final shared/static library. Is there a concern that because `gboolean` is defined in mono's `glib.h`, that when it makes it to the end user, they wouldn't have the `glib.h` and therefore wouldn't know the `gboolean` type? Or is that not the case because the `gboolean` is understood to be `int32_t` after linking everything together when building the shared/static library?

Reply:
Why do we need the header? In the generated data/len source files we should only use simple C99 data types for byte array and len and then we only have one source file generated for the preallocated bundles and registration code, so all the struct layouts and extern declare of registration functions used by registration code can go directly into that source file, no need to expose additional information in generated header at the moment. The only thing that we might put into a generated header is the prototype of the generated registration function. Having that said, the generated source files still need to compile without to much knowledge of Mono specific headers, so all generated code should only reference C99 data types.

Reply:
Ah, I just figured that there should be an alignment of the types being used in the source files generated by the `EmitBundle` task and the types we use in the mono side. I wasn't sure if it was problematic to have "different" surface types yet same underlying type, so I erred to just keep them aligned as much as possible, keeping the generated `mono-bundled-source.h` to be a close copy of `bundled-resources-internals.h`.

If the generated source/object file does not need to have the same surface type `gboolean/bool/unsigned char/etc` as long as the underlying type `int32_t` etc then I'll remove the mono specific types and header. Moreover, the header was an artifact of having a separate source file for preallocating `MonoBundled*Resource` structs and registering aforementioned `MonoBundled*Resource` structs, but since we merged the two into one, then there is less of a need to have a header. If the only thing valuable to place in the header is the prototype of the generated registration function, wouldn't that only be beneficial if the end user needed to include that header to inform the compiler that the symbol exists? That would be an equivalent of just prototyping the function in the end user's own source file where they need to call that registration function right? Does one function justify a header?

I'll try to remove the header and change the types in the `EmitBundle` generated source file to be C99 data types, and keep the types in `bundled-resources-internals.h` more aligned with mono conventions (i.e. `gboolean` over `bool`)

Reply:
Removed all internal types to align on C99 types